contracts/StablecoinConverter.sol

```diff
@@ -108,4 +109,62 @@ contract StablecoinConverter is EpochTokenLocker {
     function tokenIdToAddressMap(uint16 id) public view returns (address) {
         return IdToAddressBiMap.getAddressAt(registeredTokens, id);
     }
-}
\ No newline at end of file
+
+    struct TradeData {
+        address owner;
+        uint volume;
+        uint16 orderId;
+    }
+
+    function submitSolution(
+        uint32 batchIndex,
+        address[] memory owner,  //tradeData is submitted as arrays
```

Comment:
I think it would be nice to pass structs of TradeData here. Managing separate arrays which elements have to align is much more messy than having one array of coherent items. Is this possible?

This would also reduce the numbers of variables you are working with.

Reply:
Completely agree. But this is only available with the experimental abiEncoder and I do not want to rely on it